astropy/cosmology/core.py

```diff
@@ -1580,6 +1576,64 @@ def inv_efunc(self, z):
 
         return 1.0 / np.sqrt(zp1 ** 2 * ((Or * zp1 + Om0) * zp1 + Ok0) + Ode0)
 
+    # The stuff below here for this class is -not- something
+    # you need to overload for your own classes.  It is done
+    # purely for efficiency reasons, and results in a 10x speedup
+    # in distance calculations.
+    @staticmethod
+    def _inv_efunc_norel(z, Om0, Ode0, Ok0):
+        opz = 1.0 + z
+        return 1.0 / (opz ** 2 * (opz * Om0 + Ok0) + Ode0)**0.5
+
+    @staticmethod
+    def _inv_efunc_nomnu(z, Om0, Ode0, Ok0, Or0):
+        opz = 1.0 + z
+        return 1.0 / ((((opz * Or0 + Om0) * opz) + Ok0) * opz**2 + Ode0)**0.5
+
+    @staticmethod
+    def _inv_efunc(z, Om0, Ode0, Ok0, Ogamma0, nufunc):
+        Or0 = Ogamma0 * (1. + nufunc(z))
+        opz = 1.0 + z
+        return 1.0 / ((((opz * Or0 + Om0) * opz) + Ok0) * opz**2 + Ode0)**0.5
+
+    def comoving_distance(self, z):
+        """ Comoving line-of-sight distance in Mpc at a given
+        redshift.
+
+        The comoving distance along the line-of-sight between two
+        objects remains constant with time for objects in the Hubble
+        flow.
+
+        Parameters
+        ----------
+        z : array-like
+          Input redshifts.  Must be 1D or scalar.
+
+        Returns
+        -------
+        d : ndarray, or float if input scalar
+          Comoving distance in Mpc to each input redshift.
+        """
+
+        from scipy.integrate import quad
+        if self._Tcmb0.value == 0:
```

Comment:
@mhvk -- what would you think about the alternative of just storing a reference to the scalar inv_efunc in, say, `self._scalar_inv_efunc` on construction?  It's not like we don't know everything we need to already at that point.